### ISSUE TITLE
Component inside the ng template not rendered

### ISSUE THREAD

**Issue by SathishKumarRajendran** [angular/angular#41371]
While running the application, it working fine but run the ng test command, the component inside the ng template not rendered .

ng serve:

I have added `<ejs-textbox>` directive inside the `<ng-template>` and it is rendered as expected while serving,
  
![image](https://user-images.githubusercontent.com/53341125/112261911-b386c580-8c92-11eb-931a-ae79a8353432.png)

code: 
```html
<e-accordionitem expanded="true">
      <ng-template #header>
        <div>Textbox</div>
      </ng-template>
      <ng-template #content>
        <app-dynamic></app-dynamic>
        <ejs-textbox placeholder="First Name" floatLabelType="Auto"></ejs-textbox>
      </ng-template>
    </e-accordionitem>
```

ng test:

While run ng test command, `<ejs-textbox>` not rendered,

![image](https://user-images.githubusercontent.com/53341125/112262241-5b03f800-8c93-11eb-8e2d-7810a7806464.png)

## 🔬 Minimal Reproduction

**sample:** https://github.com/SathishKumarRajendran/ngtemplateissue

**Steps:**
* npm i

working: 
* ng serve

Not rendered: 
* ng test

## 🔥 Exception or Error

I didn't get any console error form `<ejs-textbox>`.

## 🌍 Your Environment
<pre><code>
Angular CLI: 11.2.5
Node: 10.15.1
OS: win32 x64

Angular: 11.2.6
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.5
@angular-devkit/build-angular   0.1102.5
@angular-devkit/core            11.2.5
@angular-devkit/schematics      11.2.5
@angular/cli                    11.2.5
@schematics/angular             11.2.5
@schematics/update              0.1102.5
rxjs                            6.6.6
typescript                      4.1.5

</code></pre>

**Comment by SathishKumarRajendran** [comment]
**Another scenario**:

While rendering the component dynamically using `ComponentFactoryResolver`, the ng template is  not working  properly as same in the unit test,

refer the below stackblitz sample,

https://stackblitz.com/edit/angular-cqbaco-4fos3j?file=dynamic-dialog%2Fdialog.service.ts

**Comment by jelbourn** [comment]
@SathishKumarRajendran could you clarify what you're trying to do? The reproduction provided has a lot going on, so providing a simpler StackBlitz would make it easier to look at

**Comment by atscott** [comment]
I'm going to close this one since the issue appears to be with the syncfusion library. Please reach out to that team for support.

**Comment by SathishKumarRajendran** [comment]
@jelbourn 

Here I have tried to use `ComponentFactoryResolver` to Retrieve the factory object that creates a component of the given type .

Then created new component [**from existing textbox compoenent**] using `create` method after i have append that newly created component to DOM like below which dont have elements inside  `ng-template`.
 
```
dynamicComponent<T>(content: Content<T>) {
    const factory = this.componentFactoryResolver.resolveComponentFactory(
      TextBoxComponent
    );
    const ngContent = this.resolveNgContent(content);
    this.dynamicComponentRef = factory.create(this.injector, ngContent);
    this.dynamicComponentRef.hostView.detectChanges();
    debugger;
    const { nativeElement } = this.dynamicComponentRef.location;
    this.document.body.appendChild(nativeElement);
  }
```

**sample:** https://github.com/SathishKumarRajendran/issueSample

Steps to reproduce issue,

- npm i 
- ng serve
- Click button to load component dynamically

### element inside ng-template renders properly without ComponentFactoryResolver

![image](https://user-images.githubusercontent.com/53341125/114155707-6582ea80-993f-11eb-8d3e-92f135c32d3f.png)





### Not get rendered when using ComponentFactoryResolver to create new component and append that in DOM,

![image](https://user-images.githubusercontent.com/53341125/114155755-77648d80-993f-11eb-9099-e168103c3b32.png)

**Comment by SathishKumarRajendran** [comment]
@jelbourn Any suggestion reg this?

**Comment by alan-agius4** [comment]
@kumaresan-subramani, I suggest that you try to reproduce the issue without any 3rd party dependencies, once that is done open a new issue including the reproduction.

Unfortunately we cannot debug complex reproductions which contains a number of 3rd party dependencies will require a lot of time.
